src/Compilers/Test/Resources/Core/TestResources.cs

```diff
@@ -20,11 +20,17 @@ public static class AssemblyLoadTests
         private static byte[] s_beta;
         public static byte[] Beta => ResourceLoader.GetOrCreateResource(ref s_beta, "AssemblyLoadTests.Beta.dll");
 
-        private static byte[] s_delta;
-        public static byte[] Delta => ResourceLoader.GetOrCreateResource(ref s_delta, "AssemblyLoadTests.Delta.dll");
+        private static byte[] s_delta1;
+        public static byte[] Delta1 => ResourceLoader.GetOrCreateResource(ref s_delta1, "AssemblyLoadTests.Delta.dll");
 
         private static byte[] s_gamma;
         public static byte[] Gamma => ResourceLoader.GetOrCreateResource(ref s_gamma, "AssemblyLoadTests.Gamma.dll");
+
+        private static byte[] s_delta2;
+        public static byte[] Delta2 => ResourceLoader.GetOrCreateResource(ref s_delta2, "AssemblyLoadTests.Version2.Delta.dll");
```

Comment:
Rather than checking in the new binaries why not compile them as a part of executing the tests? That is generally more maintainable and doesn't contribute to the number of checked in binaries that we have.

Reply:
I felt stumped on how to do this for a bit, but it feels like it might be reasonable to just compile+emit them during the test with the compiler APIs..? Do you have a preference?

Reply:
That's roughly what I had in mind. Just use the existing `CreateCompilation` helpers then emit the images to disk.